pigpen-avro/src/main/clojure/pigpen/avro/core.clj

```diff
@@ -0,0 +1,120 @@
+;;
+;;
+;;  Copyright 2013 Netflix, Inc.
+;;
+;;     Licensed under the Apache License, Version 2.0 (the "License");
+;;     you may not use this file except in compliance with the License.
+;;     You may obtain a copy of the License at
+;;
+;;         http://www.apache.org/licenses/LICENSE-2.0
+;;
+;;     Unless required by applicable law or agreed to in writing, software
+;;     distributed under the License is distributed on an "AS IS" BASIS,
+;;     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;;     See the License for the specific language governing permissions and
+;;     limitations under the License.
+;;
+;;
+
+(ns pigpen.avro.core
+  (:require [clojure.string :as str]
+            [pigpen.raw :as raw]
+            [pigpen.core :as pig])
+  (:import [org.apache.avro
+            Schema
+            Schema$Field
+            Schema$Type
+            Schema$Parser
+            SchemaNormalization]
+           [org.apache.avro.file DataFileReader]
+           [org.apache.avro.specific SpecificDatumReader]
+           [org.apache.avro.generic GenericData$Record]
+           [pigpen.local PigPenLocalLoader]))
+
+(set! *warn-on-reflection* true)
+
+(declare field-names)
```

Comment:
Fair enough. In other places where that's the case, I'll put the logic in a different namespace and just expose the public API by copying the symbols there. Take a look at `pigpen.core` for examples. It also makes testing easier because you can access all of the internals easily.